### ISSUE TITLE
Allow `hide closed` in the Gantt chart

### ISSUE THREAD

**Issue by cilerler** [plantuml/plantuml#1807]
Allow the option to hide closed days in the Gantt chart.

**Comment by The-Lum** [comment]
Hi @cilerler,

_[Just for the record]_
See similar request here:
- https://forum.plantuml.net/17652/gantt-diagram-can-we-hide-closed-dates

Regards,
Th.

**Comment by cilerler** [comment]
Thanks, I actually saw that before I posted. I was thinking of creating a representation here, as many new folks might not be aware of that forum.

**Comment by yaohwang** [comment]
@The-Lum Do we have a plan for that? I think this little change makes a qualified improvement for presentation, maybe we should keep a little gap between weeks for distinction, but no dashes need any more.

**Comment by The-Lum** [comment]
FYI @arnaudroques

**Comment by arnaudroques** [comment]
With version 1.2025.0, you can set up your project timeline as follows.
(This is really an alpha feature, and it works only when saturday and sunday are closed)

```
@startgantt

sunday are closed
saturday are closed

hide closed

Project starts the 2024/11/07 
[Task1] on {Alice} requires 30 days
[Task2] on {Bob:50%} requires 10 days
then [Task3] on {Alice:25%} requires 10 days
then [Task4] on {Alice:25%} {Bob} requires 10 days

@endgantt
```

Is this in line with what you were expecting?

We would love to hear your thoughts and feedback before we proceed further! 😊

**Comment by The-Lum** [comment]
FYI @cilerler, @yaohwang

**Comment by cilerler** [comment]
Thanks for your effort. I'll look into it today and get back to you, @arnaudroques.  
I suggest adding the option to close a specific date _(e.g., `2025/01/01 is closed`)_ as well.

**Comment by yaohwang** [comment]
thanks for that @arnaudroques @The-Lum 

I'm using http://www.plantuml.com/plantuml/uml/SyfFKj2rKt3CoKnELR1Io4ZDoSa700001 currently, seems not have update, I'll check into that soon with version 1.2025.0 

If we can make `hide closed` also work on any closed days such as `2025-01-25 to 2025-02-04 is closed`, that would be much better for me. Thks again!

**Comment by arnaudroques** [comment]
> I'll check into that soon with version 1.2025.0
> 

It's [online now](https://editor.plantuml.com/uml/VT2n2i8m483XFKznYtEJj4NeL7q21pTnYCrXgoN1kyiWuhihgM0fufZmtNzmBOiZERiWeZW5x-xW2A4R8wDNx2JHua_rW_y-TXGls0da38Fq25RRgZ2cq6jGXxtZgpb23F3OZ4E7Jo2yfO6GeTJmJlDax6IsyTJKUZLJvgEanm2PbhDSO-l_k5hYlEF7Y6en-7o75m00)!

> If we can make `hide closed` also work on any closed days such as `2025-01-25 to 2025-02-04 is closed`, that would be much better for me. Thks again!

Ok, this is the next move! :-)

**Comment by cilerler** [comment]
I just tested it, and it's working perfectly! 
Looking forward to hiding the other closed days as well. 
Thanks again!

**Comment by yaohwang** [comment]
It worked, Thanks! 
Expecting the next move to hiding all closed days.

> Ok, this is the next move! :-)

**Comment by The-Lum** [comment]
Hi all,

To follow the tests, and feedbacks...
- Could you allow `hide closed` also for `weekly`, `monthly`, `quarterly,`...  prinstscale?

Here is an example for `weekly`:
```puml
@startgantt
printscale weekly
saturday are closed
sunday are closed

hide closed

Project starts the 1st of january 2025
[Prototype design end] as [TASK1] requires 19 days
[TASK1] is colored in Lavender/LightBlue
-- other task --
[Testing] requires 14 days
[TASK1]->[Testing]

2025-01-18 to 2025-01-22 are colored in salmon 
@endgantt

```
>[![](https://img.plantuml.biz/plantuml/svg/NL1DJyCm3Br7odzuVo1YhK222ngxiiCak4qxM8rfiuLaYrrG_pqf31WSd-ttvQKe5UqfgLfpB26fT1GPxyo7E5aZf6FnD84AeujPsDVXcFwFh1c2luIRalVSAJxr1Jem6b7a5-mfZLGcj8lsnffjFTIiqv7XMKAVmCdlG8BjyyFJOxD3uTCO2WkQErHRgPpp9WYw77DXZv2mfhTAvNAr3lsWgpYoDSuXL-i29Jd0kPdBeY7rbxhNVtNT_S_Ht6JEwHQDQswX6T-eRR_w__eBnTUSOCsonZX_z0C0)](https://editor.plantuml.com/uml/NL1DJyCm3Br7odzuVo1YhK222ngxiiCak4qxM8rfiuLaYrrG_pqf31WSd-ttvQKe5UqfgLfpB26fT1GPxyo7E5aZf6FnD84AeujPsDVXcFwFh1c2luIRalVSAJxr1Jem6b7a5-mfZLGcj8lsnffjFTIiqv7XMKAVmCdlG8BjyyFJOxD3uTCO2WkQErHRgPpp9WYw77DXZv2mfhTAvNAr3lsWgpYoDSuXL-i29Jd0kPdBeY7rbxhNVtNT_S_Ht6JEwHQDQswX6T-eRR_w__eBnTUSOCsonZX_z0C0) 

Regards,
Th.

**Comment by pboushy** [comment]
I was excited to see this had been added in 1.2025.0, but didn't realize this is only for daily.
I also want to see this for weekly.

**Comment by arnaudroques** [comment]
> I was excited to see this had been added in 1.2025.0, but didn't realize this is only for daily. I also want to see this for weekly.

It should also for for weekly. Could you post a small example that shows the issue? Thanks!

**Comment by The-Lum** [comment]
@arnaudroques: see previously example:
- https://github.com/plantuml/plantuml/issues/1807#issuecomment-2579973357

_And have only a simple dotted line, or not depending on parameter..._
Regards,
Th.

**Comment by univ-contrib** [comment]
Yes, this is very much desired

> With version 1.2025.0, you can set up your project timeline as follows. (This is really an alpha feature, and it works only when saturday and sunday are closed)
> 
> ```
> @startgantt
> 
> sunday are closed
> saturday are closed
> 
> hide closed
> 
> Project starts the 2024/11/07 
> [Task1] on {Alice} requires 30 days
> [Task2] on {Bob:50%} requires 10 days
> then [Task3] on {Alice:25%} requires 10 days
> then [Task4] on {Alice:25%} {Bob} requires 10 days
> 
> @endgantt
> ```
> 
> Is this in line with what you were expecting?
> 
> We would love to hear your thoughts and feedback before we proceed further! 😊

**Comment by arnaudrichard** [comment]
I don't know for the OP, but for me hiding closed days could be as simple as having a continuous task over them. I could then use style to make them transparent.
 But currently even when closed days are transparent the tasks look ugly:
```
@startgantt

project starts on 2025-03-31
saturday are closed
sunday are closed
printscale weekly

[Datasheet] on {Arnaud:25%} requires 4 weeks

<style>
ganttDiagram {
 closed {
  BackgroundColor transparent
 }
}
</style>

@endgantt
```

![Image](https://github.com/user-attachments/assets/b9d57600-43d2-468b-af2c-8ee0990c0387)

**Comment by prasun3** [comment]
With `hide closed`, it shows a dotted line between weeks. Is it possible to change the color/style of this line? If so, which style parameter controls this?

**Comment by chipbite** [comment]
Yes, I think for weekly, the problem is indeed how the hidden days are displayed, they make the bars hard to read and basically cant be used in any real sense.

Great thread, good potential improvement!

Thanks for an awesome tool!
